document/core/text/modules.rst

```diff
@@ -380,6 +387,56 @@ Memories can be defined as :ref:`imports <text-import>` or :ref:`exports <text-e
 The latter abbreviation can be applied repeatedly, with ":math:`\dots`" containing another import or export or an inline data segment.
 
 
+.. index:: event, event type, identifier, function type, exception attribute
+   pair: text format; event
+.. _text-event:
+
+Events
+~~~~~~
+
+An event definition can bind a symbolic :ref:`event identifier <text-id>`.
+
+.. math::
+   \begin{array}{llcl}
+   \production{event} & \Tevent_I &::=&
+     \text{(}~\text{exception}~~\Tid^?~~x,I'{:}\Ttypeuse_I~\text{)} \\ &&& \qquad
+       \Rightarrow\quad \{ \EVATTRIBUTE~\EXCEPTION, \EVTYPE~x \} \\
+   \end{array}
+
+.. note::
+   The syntax for events just covers the only possible event attribute, that of an exception.
+   In future versions of WebAssembly, more events may be added, and the above syntax
+   may become an abbreviation.
+
+
+.. index:: import, name
+   pair: text format; import
+.. index:: export, name, index, event index
+   pair: text format; export
+.. index:: exception
+.. _text-event-abbrev:
+
+Abbreviations
+.............
+
+Events can be defined as :ref:`imports <text-import>` or :ref:`exports <text-export>` inline:
+
+.. math::
+   \begin{array}{llclll}
+   \production{module field} &
+     \text{(}~\text{exception}~~\Tid^?~~\text{(}~\text{import}~~\Tname_1~~\Tname_2~\text{)}~~\Teventtype~\text{)} \quad\equiv \\ & \qquad
```

Comment:
Something seems wrong here: the `Teventtype` as described would require another `exception` keyword. Instead, there has to be a typeuse here (cf. the analogous abbrevs for functions).

Reply:
Done